src/map/src/cmapclient.cpp

```diff
@@ -54,20 +54,24 @@ bool CMapClient::HandlePacket(uint8_t* _buffer) {
       return JoinServerReply(getPacket<ePacketType::PAKCS_JOIN_SERVER_REQ>(
           _buffer));  // Allow client to connect
     case ePacketType::PAKCS_CHANGE_MAP_REQ:
+      if (login_state_ != eSTATE::LOGGEDIN) {
+        logger_->warn("Client {} is attempting to execute an action before logging in.", get_id());
+        return true;
+      }
             entity_.component<BasicInfo>()->loggedIn_.store(false);
-        default:
+            break;
+    default:
             break;
     }
+  if (login_state_ != eSTATE::LOGGEDIN) {
+    logger_->warn("Client {} is attempting to execute an action befire logging in.", get_id());
+    return CRoseClient::HandlePacket(_buffer);
```

Comment:
Please mark this line in a comment for deletion with some unique tag.  This will make it easier later down the road to find these lines and remove them.